api/src/database/migrations/20230721A-require-shares-fields.ts

```diff
@@ -1,37 +1,58 @@
+import { createInspector } from '@directus/schema';
 import type { Knex } from 'knex';
+import logger from '../../logger.js';
 
 export async function up(knex: Knex): Promise<void> {
-	await knex.schema.alterTable('directus_shares', (table) => {
-		if (knex.client.constructor.name === 'Client_MySQL') {
-			// Temporary drop foreign key constraint, see https://github.com/directus/directus/issues/19399
-			table.dropForeign('collection', 'directus_shares_collection_foreign');
-		}
+	await dropConstraint(knex);
 
+	await knex.schema.alterTable('directus_shares', (table) => {
 		table.dropNullable('collection');
-
-		if (knex.client.constructor.name === 'Client_MySQL') {
-			// Recreate foreign key constraint, from 20211211A-add-shares.ts
-			table.foreign('collection').references('directus_collections.collection').onDelete('CASCADE');
-		}
-
 		table.dropNullable('item');
 	});
+
+	await recreateConstraint(knex);
 }
 
 export async function down(knex: Knex): Promise<void> {
-	await knex.schema.alterTable('directus_shares', (table) => {
-		if (knex.client.constructor.name === 'Client_MySQL') {
-			// Temporary drop foreign key constraint, see https://github.com/directus/directus/issues/19399
-			table.dropForeign('collection', 'directus_shares_collection_foreign');
-		}
+	await dropConstraint(knex);
 
+	await knex.schema.alterTable('directus_shares', (table) => {
 		table.setNullable('collection');
+		table.setNullable('item');
+	});
 
-		if (knex.client.constructor.name === 'Client_MySQL') {
-			// Recreate foreign key constraint, from 20211211A-add-shares.ts
-			table.foreign('collection').references('directus_collections.collection').onDelete('CASCADE');
+	await recreateConstraint(knex);
+}
+
+/**
+ * Temporary drop foreign key constraint for MySQL instances, see https://github.com/directus/directus/issues/19399
+ */
+async function dropConstraint(knex: Knex) {
+	if (knex.client.constructor.name === 'Client_MySQL') {
```

Comment:
I wonder should we use the `isOneOfClients` helper used in other migrations (though in this case we're only targeting MySQL so it may feel a tad redundant), such as:

https://github.com/directus/directus/blob/d5694ec443a358536cef9e50ffa179bd3abab0ba/api/src/database/migrations/20210920A-webhooks-url-not-null.ts#L7

understandably `knex.client.constructor.name` was also technically used back in:

https://github.com/directus/directus/blob/d5694ec443a358536cef9e50ffa179bd3abab0ba/api/src/database/migrations/20210519A-add-system-fk-triggers.ts#L111

However since `isOneOfClients` seems to exist after `20210519A-add-system-fk-triggers.ts` was written, maybe we can try to use helpers in newer migrations.

Reply:
Sounds cleaner to me ❤️ Updated now 👍